src/util/string.h

```diff
@@ -214,12 +215,9 @@ inline std::string MakeUnorderedList(const std::vector<std::string>& items)
 /**
  * Check if a string does not contain any embedded NUL (\0) characters
  */
-[[nodiscard]] inline bool ContainsNoNUL(std::string_view str) noexcept
+[[nodiscard]] inline bool ContainsNUL(std::string_view str) noexcept
 {
-    for (auto c : str) {
-        if (c == 0) return false;
-    }
-    return true;
+    return std::ranges::any_of(str, [](char c) { return c == 0; });
```

Comment:
We should be able to simplify further:
```suggestion
    return str.find('\0') != std::string_view::npos;
```

Reply:
Applied in https://github.com/bitcoin/bitcoin/pull/31301/commits/6077262788333759f1ddc27539e86cd04ac8d076